### PR TITLE
[Azure Blob Storage] Add logs

### PR DESCRIPTION
**Pull request by moxarth-rathod**
## Part Of https://github.com/elastic/connectors/issues/2299

Adding more logs in Azure Blob Storage connector.

Log file: https://drive.google.com/file/d/1RMuieFYk-llbqZUj__eYJbnUY5W2wq7K/view?usp=drive_link
Updated file: https://drive.google.com/file/d/1dVEpn6ccoz5uKAgzSrahtDp1vyifJDIh/view?usp=drive_link 

#### Pre-Review Checklist
- [x] this PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `config.yml.example`)
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

File: connectors/sources/azure_blob_storage.py [hunk 1]
```diff
@@ -212,6 +216,7 @@ async def get_container(self, container_list):
         Yields:
             dictionary: Container document with name & metadata
         """
+        self._logger.debug("Fetching containers")
```
Comment (jedrazb) [inline connectors/sources/azure_blob_storage.py:219]: in `get_blob` we have log level `INFO`, maybe worth changing log level `INFO` here as well to keep it consistent?

**Review by jedrazb — COMMENTED** [review]
Looks good, one additional thing I noticed is that in `get_docs` we have 

```
 async def get_docs(self, filtering=None):
        ...
        async for container_data in self.get_container(container_list=self.containers):
            if container_data:
                ...
```

I don't have the context to know why we need `if container_data` but if it's needed I would log a `debug` statement that in `else` clause that we are skipping container because it's empty(?) / doesn't have data?

File: connectors/sources/azure_blob_storage.py [hunk 2]
```diff
@@ -185,6 +185,7 @@ async def get_content(self, blob, timestamp=None, doit=None):
         if not self.can_file_be_downloaded(file_extension, filename, file_size):
             return
 
+        self._logger.debug(f"Downloading content for file: {filename}")
```
Comment (artem-shelkovnikov) [inline connectors/sources/azure_blob_storage.py:188]: Looks like a duplicate line - you already have "f"Downloading content for blob: {blob_name} from {container_name} container"" that happens right after

File: connectors/sources/azure_blob_storage.py [hunk 3]
```diff
@@ -247,6 +252,7 @@ async def get_blob(self, container):
         Yields:
             dictionary: Formatted blob document
         """
+        self._logger.info(f"Fetching blobs for '{container['name']}' container")
```
Comment (artem-shelkovnikov) [inline connectors/sources/azure_blob_storage.py:255]:
```suggestion
        self._logger.info(f"Fetching blobs for container '{container['name']}'")
```

**Review by artem-shelkovnikov — COMMENTED** [review]
1. Would be also cool to log how many blobs are found in the container. Also would be nice to report progress in format of "Extracted X out of Y blobs for container {container}" reported every 100 lines and in the end of sync.
2. Log lines "Generating connection string." should be DEBUG
3. Log lines "Successfully connected to the Azure Blob Storage" should be DEBUG
4. Need a DEBUG log line saying which containers were found, then which ones will be synced in the end after all validations for it

**Comment by moxarth-elastic** [comment]
> I don't have the context to know why we need `if container_data` but if it's needed I would log a `debug` statement that in `else` clause that we are skipping container because it's empty(?) / doesn't have data?

This condition was added due this `if` statement. So, at the end of the execution of this method https://github.com/elastic/connectors/blob/22bfb35cbfa412b845aeb55b19f080594235af3e/connectors/sources/azure_blob_storage.py#L211 we return None so added a condition in get_docs to handle it.

**Comment by jedrazb** [comment]
@moxarth-elastic could you share updated example DEBUG output, before we hit ✅

**Comment by moxarth-elastic** [comment]
> @moxarth-elastic could you share updated example DEBUG output, before we hit ✅

Here is the updated log file: https://drive.google.com/file/d/1HJ36WDN87P3q4fCeQGu8FiQUrh9texD-/view?usp=drive_link

**Comment by praveen-elastic** [comment]
Buildkite test this

File: connectors/sources/azure_blob_storage.py [hunk 4]
```diff
@@ -280,12 +282,18 @@ async def get_blob(self, container):
...
+        blob_count = 0
         try:
             async for blob in container_client.list_blobs(include=["metadata"]):
+                blob_count += 1
                 yield self.prepare_blob_doc(
                     blob=blob, container_metadata=container["metadata"]
                 )
+            self._logger.info(
+                f"Fetched {blob_count} blobs from '{container['name']}' container"
+            )
```
Comment (artem-shelkovnikov) [inline connectors/sources/azure_blob_storage.py:296]: This should go into a `finally` statement so that it's always shown.

**Review by artem-shelkovnikov — COMMENTED** [review]
Any chance we can have logs of actual requests to the ABS?

**Comment by moxarth-elastic** [comment]
@artem-shelkovnikov added logs here: https://drive.google.com/file/d/1dVEpn6ccoz5uKAgzSrahtDp1vyifJDIh/view?usp=drive_link

**Pull request closed without merging**